client/lib/sites-list/actions.js

```diff
@@ -47,20 +47,34 @@ var SitesListActions = {
 		}
 	},
 
-	deleteSite: function( site ) {
+	deleteSite: function( site, onComplete ) {
 		Dispatcher.handleViewAction( {
 			type: 'DELETE_SITE',
 			site: site
 		} );
 
 		debug( 'Deleting site', site );
-		wpcom.deleteSite( site.ID, SitesListActions.receiveDeletedSite );
+
+		wpcom.deleteSite( site.ID, function( error, data ) {
```

Comment:
It would be so nice if we could do all this wpcom methods to return promises someday :/